### DISCUSSION TITLE
useQueries return in map order?

### DISCUSSION THREAD

**Discussion by ljones87** [TanStack/query#2624] [Q&A]
Sorry if this was within one of the discussions already answered. Feel free to just link me to it. Didnt see on the [docs page.](https://react-query.tanstack.com/reference/useQueries#_top)

Wanted to confirm that useQueries returns them in order of execution. So, for example, if I have two users in an array, I can safely destructure `const [user1, user2] = useQueries([user1, user2].map(user =>..))`

**Comment by TkDodo** [comment]
Yep, that should work. However, user2 could be in success state while user1 is still loading because the queries run in parallel

**Reply by ljones87** [reply]
Ok great. Yes understood, each query in the array would have the full return object and query state would need to be analyzed before handling. Knowing that they return in the same order though is very helpful. Thank you!

**Reply by patrickmcdougle-okta** [reply]
It'd be great if we could get the queryKey back in the query result too so we don't have to rely on index matching

**Reply by TkDodo** [reply]
that's a good point @patrickmcdougle-okta . Would you maybe like to PR that?

**Reply by patrickmcdougle-okta** [reply]
I probably can't as part of my responsibilities at work, but maybe @patrick-mcdougle would help.

**Reply by patrick-mcdougle** [reply]
https://github.com/tannerlinsley/react-query/pull/2690
